### ISSUE TITLE
[Enhancement] Stop using the RSS feeds for fetching the videos of a youtube channel

### ISSUE THREAD

**Issue by unixfox** [iv-org/invidious#2410]
Use the innertube API for fetching the videos of a youtube channel. This endpoint is not rate limited.

**Comment by syeopite** [comment]
The RSS feeds is only used to fetch videos to populate and update user feeds like subscriptions, notification and the instance-wide popular feed. Everything else is done via InnerTube.

**Comment by TheFrenchGhosty** [comment]
@Fijxu @unixfox Is it still the case? Still relevant?

**Comment by unixfox** [comment]
Yes it is. From time to time the RSS endpoints are down and this impact invidious. Recent downtime: https://github.com/iv-org/invidious/issues/5679

**Comment by TheFrenchGhosty** [comment]
A 50$ bounty has been added to this issue.

Anyone opening a PR fixing this issue, will receive 50$ (in BTC) from the Invidious project.

More details: #1898

**Comment by Fijxu** [comment]
Yep, we need to fix this :p, RSS feeds are down or rejecting requests right now (at the time of adding this comment), they are not really reliable to get channel information, that is why users sometimes can't subscribe to channels and describe this type of problems: https://github.com/iv-org/invidious/issues/5651, https://github.com/iv-org/invidious/issues/4657 (When trying to subscribe using the channel subscribe button, it reverts itself to subscribe again, since the operation fails)

**Comment by SmartMenu9872** [comment]
Hola, me interesa trabajar en este issue. ¿Sigue vigente la recompensa de 50 USD y aceptan una implementación basada en la API InnerTube? Antes de invertir tiempo en el desarrollo, ¿podrían confirmar el alcance esperado y el mecanismo de pago?

**Comment by SmartMenu9872** [comment]
Hey @TheFrenchGhosty, are you marking me as spam? If I'm right, I would like to know why.

**Comment by TheFrenchGhosty** [comment]
@SmartMenu9872 Because you are not respecting the AI policy. You're banned from the repo now.
